### DISCUSSION TITLE
Please how can I join more than more than two streams

### DISCUSSION THREAD

**Discussion by Oladeji** [LGouellec/streamiz#252] [Q&A] [answered]
Hi @All
Thanks for this library , it makes life easy for me ,
Please how can I include  stream3 or more  in the join below ?  I will appreciate any help

internal class Program
    {
        static async System.Threading.Tasks.Task Main(string[] args)
        {
                var config = new StreamConfig<StringSerDes, StringSerDes>
                {...
                };
                StreamBuilder builder = new StreamBuilder();
                string topic1 = "user-topic";
                string topic2 = "detail-topic";
                string topic3 = "jobdetails-topic"; // the new topic that i want to join
                var stream1 = builder.Stream<string, string>(topic1);
                var stream2 = builder.Stream<string, string>(topic2);
                var stream3 = builder.Stream<string, string>(topic3);
                stream1.Join(stream2, user_userdetailJoiner(), JoinWindowOptions.Of(TimeSpan.FromMinutes(1))).To("full-join");
                Topology t = builder.Build();
                KafkaStream stream = new KafkaStream(t, config);
                await stream.StartAsync();
        }

        private static Func<string, string, string> user_userdetailJoiner()
        {
            return (v1, v2) =>
            {
                User? us1 = JsonSerializer.Deserialize<User>(v1);
                UserDetails? us2 = JsonSerializer.Deserialize<UserDetails>(v2);
                var result = new UserD { key = us1.key, Address = us2.Address, Country = us2.Country, Lastname = us1.Lastname, FirstName = us1.FirstName };
                var s = JsonSerializer.Serialize<UserD>(result);
                Console.WriteLine(s);
                return s;
            };
        }
    }

**Comment by LGouellec** [comment, marked as answer]
Hi @Oladeji ,

First a join operation contains two elements, stream-table, table-table or stream-stream. 

So you can't join three streams between them (same behavior in JAVA).

But you can link multiple join like

``` csharp
var stream1 = builder.Stream(...);

var stream2 = builder.Stream(...);

var stream3 = builder.Stream(...);

stream1.Join(stream2, ....)
.Join(stream3, ....)
.To(output);

```

**Comment by Oladeji** [comment]
Hi LGouellec.Thank you for your response
Oladeji








Through God we shall do valiantly
        Psalm 60:12;Psalm 108:13 

    On Saturday, May 6, 2023 at 04:16:56 PM GMT+1, LGouellec ***@***.***> wrote:  
 
 


Hi @Oladeji ,

First a join operation contains two elements, stream-table, table-table or stream-stream.

So you can't join three streams between them (same behavior in JAVA).

But you can link multiple join like
var stream1 = builder.Stream(...);

var stream2 = builder.Stream(...);

var stream3 = builder.Stream(...);

stream1.Join(stream2, ....)
.Join(stream3, ....)
.To(output);
